### ISSUE TITLE
TypeError: undefined is not an object (evaluating 't[e+"X"]')

### ISSUE THREAD

**Issue by lookfirst** [taye/interact.js#284]
I see this error every once and a while in my code. The subject is a bit off cause my code is minified. =)

I've traced that down to this line of code in getXY:

https://github.com/taye/interact.js/blob/master/interact.js#L648

It appears that sometimes `pointer` is undefined.

Any chances of adding some undefined protection around that code?

Thanks.

**Comment by taye** [comment]
This should be fixed in v1.2.5. Which version are you using?

**Comment by lookfirst** [comment]
I am on master cause you hadn't done a release in a while. =)

Just saw another one in my rollbar logs overnight.

**Comment by taye** [comment]
Has it happened multiple times? Can you get a stack trace (even if it's minified)?

**Comment by lookfirst** [comment]
Sadly that is all I get from rollbar for this. Sorry! Wish I could help more.

**Comment by lookfirst** [comment]
Yes. It is happening a few times a day on a fairly heavy trafficked website.

**Comment by taye** [comment]
I've removed the code that splices idle interactions. That *might* help. I'm extremely reluctant to add undefined protection.

**Comment by lookfirst** [comment]
```
undefined is not an object (evaluating 'pointer[type + 'X']')
1 File bundle.js line 47721 col 19 in getXY
2 File bundle.js line 47732 col 12 in getPageXY
3 File bundle.js line 47693 col 14 in setEventXY
4 File bundle.js line 48279 col 24 in setEventXY
5 File bundle.js line 48572 col 24 in pointerMove
6 File bundle.js line 49656 col 30 in [anonymous]
```

**Comment by lookfirst** [comment]
@taye Your latest changes seem to have helped. =) I'm going to close this now. Thanks so much for your help.

**Comment by ptchand** [comment]
Observing the same issue "TypeError: undefined is not an object (evaluating 't[e+"X"]')"

**Comment by shashank100** [comment]
@taye Facing the same issue @ptchand did you find anything

**Comment by shankhadeep** [comment]
@taye I am also facing the same issue, please help us

**Comment by ptchand** [comment]
Facing few other errors related to it
- TypeError: Unable to get property 'pageX' of undefined or null reference
- TypeError: undefined is not an object (evaluating 't[e+"X"]')
- TypeError: Cannot read property 'pageX' of undefined
- TypeError: undefined is not an object (near '...},e=e||"page",n.x=t[e+"X"],n.y=t[e+"Y"],...')

**Comment by taye** [comment]
Use `npm install interactjs@next" or download it from https://rawgit.com/taye/interact.js/unstable/dist/interact.min.js.

**Comment by ptchand** [comment]
We are using Webpack. Getting this following error during UglifyJs. 

00:30:19 [1m[31mERROR in js/qbo-help-ui.js from UglifyJs
00:30:19 SyntaxError: Unexpected token: operator (>) [./~/interactjs/src/utils/isWindow.js:1,0][39m[22m
00:30:19 Child [1mextract-text-webpack-plugin[22m:

**Comment by alpritt** [comment]
I'm still getting this error in v1.2.9 in my production logs. Should this be fixed in 1.2.9 or only in the 1.3.0 alphas?

I've not been able to recreate this myself yet. But if you have any idea what could be causing it, I will try to do so.

`undefined is not an object (evaluating 't[e+"X"]')` is occurring in iOS 10.x on iPad an iPhone or various other devices are giving `Cannot read property 'pageX' of undefined`. Both are at line 650 in getXY(). The line is `xy.x = pointer[type + 'X'];`

Stacktrace:

```
TypeError: undefined is not an object (evaluating 't[e+"X"]')
  at getXY(~/interactjs/interact.js:650:15)
  at getPageXY(~/interactjs/interact.js:667:12)
  at setEventXY(~/interactjs/interact.js:611:8)
  at method(~/interactjs/interact.js:1784:21)
  at apply(~/interactjs/interact.js:3325:32)
  at d(raven.js:386:28)
```

**Comment by taye** [comment]
@alpritt I can't verify that it's fixed, but it might be worth using the newer version. I've written a quick [migration guide](https://github.com/interactjs/docs/blob/v1.3/migration-v1.2-1.3.en.html.md) to make this easier. If anything trips you up, I'd be happy to help.

**Comment by alpritt** [comment]
Thanks @taye. I had a brief try of the alpha 0 and it's causing the native browser drag to appear (this is on an `<a>` element) which I think is cancelling interact's drag behaviour. I worked around this problem before, but my tricks to stop it aren't working now for some reason. So I will come back to this when I have more time.

I also get this warning when running this through webpack (which I didn't get with 1.2.9):

```
WARNING in ./~/interactjs/dist/interact.js
Critical dependencies:
8:481-488 This seems to be a pre-built javascript file. Though this is possible, it's not recommended. Try to require the original source to get better results.
 @ ./~/interactjs/dist/interact.js 8:481-488
```
Not sure if that is me installing it incorrectly. I just did `sudo npm install git+https://git@github.com/taye/interact.js#v1.3.0-alpha.0`
